tools/cli/commands/create.py

```diff
@@ -358,80 +383,108 @@ def flags(parser):
     Raises:
       subprocess.CalledProcessError: If the `gcloud` command fails
     """
-    get_cmd = ['networks', 'describe', '--format', 'value(name)',
-               _DATALAB_NETWORK]
+    get_cmd = ['networks', 'describe', '--format', 'value(name)', network_name]
     try:
         utils.call_gcloud_quietly(
             args, gcloud_compute, get_cmd, report_errors=False)
     except subprocess.CalledProcessError:
-        create_network(args, gcloud_compute)
+        create_network(args, gcloud_compute, network_name)
     return
 
 
-def create_firewall_rule(args, gcloud_compute):
-    """Create the `datalab-network-allow-ssh` firewall rule.
+def create_firewall_rule(args, gcloud_compute, network_name, rule_name):
+    """Create the `{rule_name}` firewall rule.
 
     Args:
       args: The Namespace returned by argparse
       gcloud_compute: Function that can be used for invoking `gcloud compute`
+      network_name: The name of the network on which to allow SSH access
+      rule_name: The name of the firewall rule
     Raises:
       subprocess.CalledProcessError: If the `gcloud` command fails
     """
     if utils.print_info_messages(args):
-        print('Creating the firewall rule {0}'.format(_DATALAB_FIREWALL_RULE))
+        print('Creating the firewall rule {0}'.format(rule_name))
     create_cmd = [
-        'firewall-rules', 'create', _DATALAB_FIREWALL_RULE,
+        'firewall-rules', 'create', rule_name,
         '--allow', 'tcp:22',
-        '--network', _DATALAB_NETWORK,
+        '--network', network_name,
         '--description', _DATALAB_FIREWALL_RULE_DESCRIPTION]
     utils.call_gcloud_quietly(args, gcloud_compute, create_cmd)
     return
 
 
-def ensure_firewall_rule_exists(args, gcloud_compute):
-    """Create the `datalab-network-allow-ssh` firewall rule if it necessary.
+def has_non_standard_firewall_rules(args, gcloud_compute, network_name):
+    rule_name = _DATALAB_FIREWALL_RULE_TEMPLATE.format(network_name)
+    list_cmd = [
+        'firewall-rules', 'list',
+        '--filter', 'network~.^*{0}$'.format(network_name),
+        '--format', 'value(name)']
+    with tempfile.TemporaryFile() as tf:
+        gcloud_compute(args, list_cmd, stdout=tf)
+        tf.seek(0)
+        matching_rules = tf.read().strip()
+        if matching_rules and (matching_rules != rule_name):
+            return True
+    return False
+
+
+def prompt_on_non_standard_firewall_rules(args, gcloud_compute, network_name):
+    if has_non_standard_firewall_rules(args, gcloud_compute, network_name):
+        print(_DATALAB_FIREWALLS_WARNING_TEMPLATE.format(network_name))
+        resp = read_input('Do you still want to use this network? (y/[n]): ')
+        if len(resp) < 1 or (resp[0] != 'y' and resp[0] != 'Y'):
+            raise CancelledException()
+    return
+
+
+def ensure_firewall_rule_exists(args, gcloud_compute, network_name):
+    """Create the `{network_name}-allow-ssh` firewall rule if it necessary.
```

Comment:
Looks like a word is missing in this comment.

Reply:
Done.